### PR TITLE
SIMPLY-3493: Redesign the age gate for the SimplyE Collection to accommodate date entry

### PR DESCRIPTION
**Pull request by MalcolmMcFly**
**What's this do?**
[Description of what this PR does goes here]

**Why are we doing this? (w/ JIRA link if applicable)**
[SIMPLY-3493
[Android] Redesign the age gate for the SimplyE Collection to accommodate date entry](https://jira.nypl.org/browse/SIMPLY-3493)

**How should this be tested? / Do these changes have associated tests?**
[Description of any tests that need to be performed once merged goes here]

**Dependencies for merging? Releasing to production?**
[Description of any watchouts, dependencies, or issues we should be aware of goes here]

**Has the application documentation been updated for these changes?**

**Did someone actually run this code to verify it works?**

### CODE REVIEW

File: simplified-ui-catalog/src/main/java/org/nypl/simplified/ui/catalog/CatalogFragmentFeed.kt [hunk 1]
```diff
@@ -473,7 +461,10 @@ class CatalogFragmentFeed : Fragment() {
   ) {
     this.uiThread.checkIsUIThread()
 
-    this.feedCOPPAGate.visibility = View.VISIBLE
+//    if (configurationService.showAgeGateUi &&
+//      this.profilesController.profileCurrent().preferences().dateOfBirth == null) {
+//        showAgeGate()
+//    }
     this.feedEmpty.visibility = View.INVISIBLE
```
Comment (io7m) [inline simplified-ui-catalog/src/main/java/org/nypl/simplified/ui/catalog/CatalogFragmentFeed.kt:468]: :nauseated_face:

File: simplified-ui-catalog/src/main/java/org/nypl/simplified/ui/catalog/CatalogFragmentFeed.kt [hunk 2]
```diff
@@ -523,29 +496,28 @@ class CatalogFragmentFeed : Fragment() {
...
+//      description.preferences.copy(dateOfBirth = this.synthesizeDateOfBirth(years))
+//    return description.copy(preferences = newPreferences)
+//  }
+//
+//  private fun synthesizeDateOfBirth(years: Int): ProfileDateOfBirth {
+//    return ProfileDateOfBirth(
+//      date = DateTime.now().minusYears(years),
+//      isSynthesized = true
+//    )
+//  }
```
Comment (io7m) [inline simplified-ui-catalog/src/main/java/org/nypl/simplified/ui/catalog/CatalogFragmentFeed.kt:513]: :nauseated_face:

**Review by io7m — CHANGES_REQUESTED** [review]
You left some things commented out, but it's otherwise all good!

File: simplified-ui-catalog/src/main/java/org/nypl/simplified/ui/catalog/CatalogFragmentFeed.kt (hunk 1 above)
Comment (MalcolmMcFly) [reply simplified-ui-catalog/src/main/java/org/nypl/simplified/ui/catalog/CatalogFragmentFeed.kt:468]: oh forgot to remove this

**Pull request merged**